### PR TITLE
Remove addresses from cluster and takem off servers

### PR DESCRIPTION
**Pull request by p-mongo**

### CODE REVIEW

File: lib/mongo/server.rb [hunk 1]
```diff
@@ -173,12 +173,27 @@ def initialize(address, cluster, monitoring, event_listeners, options = {})
       @address = address
       @cluster = cluster
       @monitoring = monitoring
+      options = options.dup
+      monitor = options.delete(:monitor)
       @options = options.freeze
+      @monitor = Monitor.new(address, event_listeners, options.merge(
+        app_metadata: cluster.app_metadata))
+      unless monitor == false
```
Comment (saghm) [inline lib/mongo/server.rb]: Similar to the Mongoid pull request from the other day, can we document that `monitor: nil` in the options is different than `monitor: false`?

Comment (p-mongo) [reply lib/mongo/server.rb]: This option is not documented because it is not meant for public consumption.

**Comment by p-mongo** [comment]
Which typo are you referring to?

**Comment by saghm** [comment]
Apologies, I was looking at the pull request title, not the commit message

**Pull request merged**